clang/lib/CIR/CodeGen/CIRGenBuiltinX86.cpp

```diff
@@ -278,6 +278,113 @@ static mlir::Value emitX86MaskTest(CIRGenBuilderTy &builder, mlir::Location loc,
   return emitIntrinsicCallOp(builder, loc, intrinsicName, resTy,
                              mlir::ValueRange{lhsVec, rhsVec});
 }
+static mlir::Value emitX86MaskedCompareResult(CIRGenFunction &cgf,
```

Comment:
Can you add a `CIRGenBuilderTy &builder` to all of these helper functions and a comment saying that the `cgf` parameter should be removed when all the NYI cases are implemented? Then replace `cgf.getBuilder()` with `builder` everywhere.